tools/dds/dds-server/lrs-device-controller.cpp

```diff
@@ -2,99 +2,205 @@
 // Copyright(c) 2022 Intel Corporation. All Rights Reserved.
 
 #include "lrs-device-controller.h"
+
 #include <librealsense2/utilities/easylogging/easyloggingpp.h>
+
+#include <realdds/dds-device-server.h>
+#include <realdds/dds-stream-server.h>
+
 #include <algorithm>
 #include <iostream>
 
+using nlohmann::json;
+using namespace realdds;
 using namespace tools;
 
-// This is a wrapper around a RealSense sensor, automatically opening and streaming it in the ctor
-// and stopping/closing on destruction
-class lrs_device_controller::lrs_sensor_streamer
+
+std::shared_ptr< dds_stream_profile > create_dds_stream_profile( rs2_stream type, nlohmann::json const & j )
 {
-public:
-    lrs_sensor_streamer( rs2::sensor rs2_sensor,
-                    rs2::stream_profile stream_profile,
-                    frame_callback_type cb )
-        : _rs2_sensor( rs2_sensor )
-        , _stream_profile( stream_profile )
-        , _frame_callback(std::move( cb ))
+    switch ( type )
     {
-        _rs2_sensor.open( _stream_profile );
-        _rs2_sensor.start( [&]( rs2::frame f ) 
-            {
-                _frame_callback( f );
-            } );
-        std::cout << _stream_profile.stream_name() << " stream started"  << std::endl;
+    case RS2_STREAM_DEPTH:
+    case RS2_STREAM_COLOR:
+    case RS2_STREAM_INFRARED:
+    case RS2_STREAM_FISHEYE:
+    case RS2_STREAM_CONFIDENCE:
+        return std::dynamic_pointer_cast< dds_stream_profile >( dds_stream_profile::from_json< dds_video_stream_profile>( j ) );
+    case RS2_STREAM_GYRO:
+    case RS2_STREAM_ACCEL:
+    case RS2_STREAM_POSE:
+        return std::dynamic_pointer_cast< dds_stream_profile >( dds_stream_profile::from_json< dds_motion_stream_profile >( j ) );
     }
-    ~lrs_sensor_streamer()
+
+    throw std::runtime_error( "Unsupported stream type" );
+}
+
+rs2_stream stream_name_to_type( std::string const & type_string )
```

Comment:
Shouldn't it be `stream_type_to_rs2`? I.e., why is it dependent on the stream name and not the stream type?

Reply:
Not for this PR. Will change when profiles are stored in stream_server

Reply:
Not sure what you mean... the profiles are there.